Modal.js

```diff
@@ -0,0 +1,172 @@
+            parsedContent.getElementById('modal-input').parentNode.removeChild(modalInput)
+
+        }
+
+        const modalPrimaryButton = parsedContent.getElementById('modal-primary-button')
+        modalPrimaryButton.innerText = this.primaryBtnText
+
+        const modalSecondaryButton = parsedContent.getElementById('modal-secondary-button')
+        modalSecondaryButton.innerText = this.secondaryBtnText
+
+        const modalDiv = parsedContent.querySelector('.modal')
+        document.body.appendChild(modalDiv)
+
+        // Event handling to close the modal with the mouse.
+        document.querySelectorAll('.modal-background, .modal-close, .modal-card-head .delete, .modal-card-foot, .exit').forEach(($close) => {
+
+            const $target = $close.closest('.modal')
+
+            $close.addEventListener('click', () => { deleteModal($target) })
+
+        }) 
+
+        // Event handling to close the modal with the keyboard.
+        document.addEventListener('keydown', (event) => { event.key === "Escape" ? closeAllModals() : undefined }) 
+
+    }
+
+    prompt(userFunction = undefined, alternateFunction = undefined) {
+
+        this.#createTemplate()
+
+        const primaryButton = document.getElementById('modal-primary-button')
+        const secondaryButton = document.getElementById('modal-secondary-button')
+        const userInput = document.getElementById('modal-input')
+        
+        primaryButton.addEventListener('click', () => {
+
+            if (userFunction) {
+
+                return userFunction(userInput.value)
+
+            }
+            
+        })
+
+        secondaryButton.addEventListener('click', () => {
+
+            if (alternateFunction) {
+
+                return alternateFunction(userInput.value)
+
+            }
+            
+        })
+
+    }
+
+    confirm(userFunction = undefined, alternateFunction = undefined) {
+
+        this.#createTemplate()
+
+        const primaryButton = document.getElementById('modal-primary-button')
+        const secondaryButton = document.getElementById('modal-secondary-button')
+
+        primaryButton.addEventListener('click', () => {
+
+            if (userFunction) {
+
+                userFunction(true)
+
+
+            }
+
+        })
+
+        secondaryButton.addEventListener('click', () => {
+
+            if (alternateFunction) {
+
+                alternateFunction(false)
+
+            }
+
+        })
+
+    }
+
+}
+
+// Removes a modal from the dom
+export function deleteModal(element, id = undefined) {
+
+  const modalToDelete = id ? document.getElementById(id) : element
+
+  modalToDelete.remove()
+
+}
+
+// Functions to open and close a modal
+export function openModal($el) {
+
+  $el.classList.add('is-active')
+
+}
+
+export function closeModal($el) {
+
+  $el.classList.remove('is-active')
+
+}
+
+export function closeAllModals() {
+
+  (document.querySelectorAll('.modal') || []).forEach(($modal) => {
+
+    deleteModal($modal);
+
+  })
+
+}
```

Comment:
I have included a selection point to prevent a modal from being created while an existing one persists. Instead, any modals are removed from the dom before proceeding to create a new modal.
A private id is now added to the class that is now included in the modal html template string.
Method names have reflected the aforemention changes and all methods are now included inside the class. Import syntax has been changed to reflect this.